src/rk.jl

```diff
@@ -0,0 +1,295 @@
+# This file contains the implementation of explicit Runkge-Kutta
+# solver from (Hairer & Wanner 1992 p.134, p.165-169).
+
+using Iterators
+
+# include the Butcher tableaus.
+include("tableaus.jl")
+
+####################
+# Iterator methods #
+####################
+
+# common structures and functions
+
+type TempArrays
+    y; ks; yerr
+end
+
+type State
+    t; dt; y; tmp :: TempArrays
+    timeout
+end
+
+
+immutable Problem{MethodType}
+    method
+    F
+    y0
+    t0
+    dt0
+    tstop
+    reltol
+    abstol
+    minstep
+    maxstep
+end
+
+
+# overload the call method for TableauRKExplicit, it returns the
+# iterator (fixed or variable step according to the tableau)
+function call(tab::TableauRKExplicit,
```

Comment:
I'd rather have an explicitly named function than overload `call`, e.g. `solver(tab, ...)` or even `solverfor(tab, ...)`.

I also think this should be considered an internal API, and we shouldn't export it. The canonical way to construct a solver should be to call `ode`, `ode4`, `ode45` etc.